### PR TITLE
Support `StringView` and `BinaryView` in CDataInterface 

### LINKED ISSUE #6170: Support BinaryView Types in C Schema FFI
**Describe the bug**

Continuation of #6163 

When attempting to bridge arrow-rs to PyArrow, seeing the following error:

```
E       pyarrow.lib.ArrowException: C Data interface error: The datatype "BinaryView" is still not supported in Rust implementation
```

### PR DESCRIPTION
**Pull request by a10y**
# Which issue does this PR close?

Closes #6170 

# Rationale for this change

Get C FFI schema conversions for view types working
 
# What changes are included in this PR?

Implement conversions in C FFI for Utf8View and BinaryView

# Are there any user-facing changes?

Schemas with views can now be serialized over the C FFI

### CODE REVIEW

File: arrow-schema/src/ffi.rs [hunk 1]
```diff
@@ -390,8 +390,10 @@ impl TryFrom<&FFI_ArrowSchema> for DataType {
             "e" => DataType::Float16,
             "f" => DataType::Float32,
             "g" => DataType::Float64,
+            "vz" => DataType::BinaryView,
```
Comment (alamb) [inline arrow-schema/src/ffi.rs:393]: I wouble checked that `vz` and `vu` are the correct format strings: https://arrow.apache.org/docs/format/CDataInterface.html#data-type-description-format-strings ✅

**Review by alamb — APPROVED** [review]
Thank you @a10y  -- looks great

**Comment by ion-elgreco** [comment]
@alamb @a10y when is this going to land in a release? :)

**Comment by alamb** [comment]
This change will be included in https://github.com/apache/arrow-rs/issues/6016

**Pull request merged**